ui/page/settingsNotifications/view.jsx

```diff
@@ -0,0 +1,181 @@
+
+    Lbryio.call('user_email', 'status', lbryIoParams)
+      .then(res => {
+        const enabledEmails =
+          res.emails &&
+          Object.keys(res.emails).reduce((acc, email) => {
+            const isEnabled = res.emails[email];
+            return [...acc, { email, isEnabled }];
+          }, []);
+
+        setTagMap(res.tags);
+        setEnabledEmails(enabledEmails);
+      })
+      .catch(e => {
+        setError(true);
+      });
+  }, []);
+
+  function handleChangeTag(name, newIsEnabled) {
+    const tagParams = newIsEnabled ? { add: name } : { remove: name };
+
+    Lbryio.call('user_tag', 'edit', { ...lbryIoParams, ...tagParams }).then(() => {
+      const newTagMap = { ...tagMap };
+      newTagMap[name] = newIsEnabled;
+
+      setTagMap(newTagMap);
+    });
+  }
+
+  function handleChangeEmail(email, newIsEnabled) {
+    Lbryio.call('user_email', 'edit', {
+      email: email,
+      enabled: newIsEnabled,
+      ...lbryIoParams,
+    })
+      .then(() => {
+        const newEnabledEmails = enabledEmails
+          ? enabledEmails.map(userEmail => {
+              if (email === userEmail.email) {
+                return { email, isEnabled: newIsEnabled };
+              }
+
+              return userEmail;
+            })
+          : [];
+
+        setEnabledEmails(newEnabledEmails);
+      })
+      .catch(e => {
+        setError(true);
+      });
+  }
+
+  if (!isAuthenticated && !verificationToken) {
+    return <Redirect to={`/$/${PAGES.AUTH_SIGNIN}?redirect=${location.pathname}`} />;
+  }
+
+  return (
+    <Page>
+      {error ? (
+        <Yrbl
+          type="sad"
+          title={__('Uh Oh')}
+          subtitle={
+            <>
+              <div>{__('There was an error displaying this page.')}</div>
+              <div className="section__actions">
+                <Button
+                  button="secondary"
+                  label={__('Refresh')}
+                  icon={ICONS.REFRESH}
+                  onClick={() => window.location.reload()}
+                />
+                <Button button="secondary" label={__('Go Home')} icon={ICONS.HOME} navigate={'/'} />
+              </div>
+            </>
+          }
+        />
+      ) : (
+        <div className="card-stack">
+          {/* @if TARGET='app' */}
+          <label className="section help">{__('App Notifications')}</label>
+          <div className="section">
+            <FormField
+              type="checkbox"
+              name="desktopNotification"
+              onChange={() => setClientSetting(SETTINGS.OS_NOTIFICATIONS_ENABLED, !osNotificationsEnabled)}
+              checked={osNotificationsEnabled}
+              label={__('Show Desktop Notifications')}
+              helper={__('Get notified when a publish or channel is confirmed.')}
+            />
+          </div>
+          {/* @endif */}
+
+          {enabledEmails && enabledEmails.length > 0 && (
+            <Card
+              title={__('Receiving Addresses')}
```

Comment:
many people will have one email, probably worth making this conditional

Reply:
That's a good point. Will change.